test/test-lib/device-type.ts

```diff
@@ -33,11 +28,11 @@ export const loadDefaultFixtures = () =>
 						options: {
 							$select: ['id', 'slug', 'name'],
 						},
-					})) as DeviceType;
+					})) as Pick<DeviceType, 'id' | 'slug' | 'name'>;
 					obj[slug] = deviceType;
 				}
 
 				return obj[slug as any];
 			},
-		}) as any as Dictionary<Promise<DeviceType>>,
+		}) as any as Dictionary<Promise<Pick<DeviceType, 'id' | 'slug' | 'name'>>>,
```

Comment:
Can't we go with
```
new Proxy({} as Dictionary<Promise<Pick<DeviceType, 'id' | 'slug' | 'name'>>, {
```

And then 
```
obj[slug] = Promise.resolve(deviceType);
```

So that all this `as any as ....` mess goes away? (worked for me locally)